src/pages/task/taskTabs/buildBaronAndAnnotations/AddIssueModal/state.ts

```diff
@@ -0,0 +1,95 @@
+import { useReducer } from "react";
+import { useSpruceConfig } from "hooks";
+import { numbers } from "utils";
+import { validateJiraURL } from "utils/validators";
+
+const { toDecimal } = numbers;
+
+interface AddIssueState {
+  url: string;
+  issueKey: string;
+  canSubmit: boolean;
+  isURLValid: boolean;
+  isKeyValid: boolean;
+  confidenceScore: string;
+}
+
+type Action =
+  | { type: "reset" }
+  | { type: "setUrl"; url: string; jiraURL: string }
+  | { type: "setKey"; issueKey: string }
+  | { type: "setConfidenceScore"; confidenceScore: string };
+
+const init = (): AddIssueState => ({
+  url: "",
+  issueKey: "",
+  canSubmit: false,
+  isURLValid: false,
+  isKeyValid: false,
+  confidenceScore: null,
+});
+
+const reducer = (state: AddIssueState, action: Action) => {
+  switch (action.type) {
+    case "reset":
+      return init();
+    case "setUrl": {
+      const isURLValid = validateJiraURL(action.jiraURL, action.url);
+      return {
+        ...state,
+        url: action.url,
+        canSubmit: isURLValid && state.isKeyValid,
+        isURLValid,
+      };
+    }
+    case "setKey": {
+      const isKeyValid = action.issueKey.length > 0;
+      return {
+        ...state,
+        issueKey: action.issueKey,
+        isKeyValid,
+        canSubmit: isKeyValid && state.isURLValid,
+      };
+    }
+    case "setConfidenceScore": {
+      const isValid =
+        !Number.isNaN(action.confidenceScore) &&
```

Comment:
If I enter something that's not a number, it let's me save it and then just shows the score as zero. 
![image](https://user-images.githubusercontent.com/13104717/167198181-c2fd9c1f-3caf-40a4-9278-f59bf8917701.png)

Reply:
![valid](https://user-images.githubusercontent.com/13104717/169355809-13e0d39e-9418-47de-a38c-951be95eac38.gif)

Reply:
also, this PR removes the validation that makes sure that there is display text defined. We shouldn't do that until after [EVG-16842](https://jira.mongodb.org/browse/EVG-16842)

Reply:
<img width="608" alt="image" src="https://user-images.githubusercontent.com/13104717/169834635-dc0f5a1e-3e52-4ab6-88dc-f7f205494b19.png">
Ideally, this shouldn't show the warning text immediately, only when the user enters something invalid. 
I'm fine with this being done in a separate ticket though if you want to get this PR closed already.